examples/preprocessing/contralateral_referencing.py

```diff
@@ -0,0 +1,71 @@
+"""
+.. _ex-contralateral-referencing:
+
+=======================================
+Using contralateral referencing for EEG
+=======================================
+
+Instead of using a single reference electrode for all channels, some
+researchers reference the EEG electrodes in each hemisphere to an electrode in
+the contralateral hemisphere (often an electrode over the mastoid bone; this is
+common in sleep research for example). Here we demonstrate how to set a
+contralateral EEG reference.
+"""
+
+import numpy as np
+import mne
+
+ssvep_folder = mne.datasets.ssvep.data_path()
+ssvep_data_raw_path = (ssvep_folder / 'sub-02' / 'ses-01' / 'eeg' /
+                       'sub-02_ses-01_task-ssvep_eeg.vhdr')
+raw = mne.io.read_raw(ssvep_data_raw_path, preload=True)
+_ = raw.set_montage('easycap-M1')
+
+# %%
+# The electrodes TP9 and TP10 are near the mastoids so we'll use them as our
+# contralateral reference channels. Then we'll create our hemisphere groups.
+
+raw.rename_channels({
+    'TP9': 'M1',
+    'TP10': 'M2'
+})
+
+# this splits electrodes into 3 groups; left, midline, and right
+ch_indices = mne.channels.make_1020_channel_selections(raw.info)
+
+# convert indices to names
+orig_names = np.array(raw.ch_names)
+ch_names = {key: orig_names[idxs].tolist() for key, idxs in ch_indices.items()}
```

Comment:
Feature request: Allow retrieval of channel names (not only indices) via `make_1020_channel_selections()`, e.g. by adding a new parameter `return_ch_names`

Reply:
https://github.com/mne-tools/mne-python/issues/11597